sdktests/server_side_events_identify.go

```diff
@@ -22,15 +22,17 @@ func doServerSideIdentifyEventTests(t *ldtest.T) {
 	events := NewSDKEventSink(t)
 	client := NewSDKClient(t, dataSource, events)
 
-	t.Run("normal user", func(t *ldtest.T) {
-		user := users.NextUniqueUser()
-		client.SendIdentifyEvent(t, user)
-		client.FlushEvents(t)
-		payload := events.ExpectAnalyticsEvents(t, defaultEventTimeout)
-		m.In(t).Assert(payload, m.Items(
-			EventIsIdentifyEvent(mockld.SimpleEventUser(user)),
-		))
-	})
+	for _, isAnonymousUser := range []bool{false, true} {
+		t.Run(selectString(isAnonymousUser, "anonymous user", "non-anonymous user"), func(t *ldtest.T) {
```

Comment:
This is just an idea, but I wonder once we have access to generics if we can make this less verbose. Something like:
```golang
func RunScenarios[Value any](t *ldtest.T, domain map[string]Value, do func (t *ldtest.T, value Value)) {
     for label, value := range domain {
          t.Run(label, func(t *ldtest.T) { do(t, value) })
     }
}

func TestXYZ(t *ldtest.T) {
     RunScenarios(t, 
          map[string]bool{"anonymous": true, "non-anonymous" : false}, 
          func(t *ldtest.T, value bool) {
               //do stuff with value
          }
     )
}
````
I'm sure there will be a bunch of generics-based testing libraries in the coming months 😵‍💫

Reply:
Yeah, even though it'll be a while before we can use generics in the SDK or in any support libraries that get compiled with the SDK, I'm hoping that we can start leveraging them in stand-alone code like this.